talisker/gunicorn.py

```diff
@@ -179,6 +193,22 @@ def init(self, parser, opts, args):
 
         return cfg
 
+    def load_config(self):
+        super(TaliskerApplication, self).load_config()
+
+        # Use pip to find out if prometheus_client is available, as
+        # importing it here would break multiprocess metrics
+        if (util.pkg_is_installed('prometheus-client') and
+                (self.cfg.workers or 1) > 1):
+            if 'prometheus_multiproc_dir' not in os.environ:
+                logger.info('running in multiprocess mode but '
+                            '`prometheus_multiproc_dir` envvar not set')
+                tmpdir = tempfile.mkdtemp()
+                os.environ['prometheus_multiproc_dir'] = tmpdir
```

Comment:
This is great - gives an ok fall back in dev or incase of misconfiguration, thanks!